rfcs/0060-content-addressed-paths.md

```diff
@@ -0,0 +1,281 @@
+- We build `contentAdressed.drv`.
+  - We first compute `dynamic(contentAdressed.drv)` to replace its
+    inputs by their real output path. Since there is none, we
+    have here `dynamic(contentAdressed.drv) == contentAdressed.drv`
+  - We realise `dynamic(contentAdressed.drv)`. This gives us an output path
+    `out(dynamic(contentAdressed.drv))`
+  - We move `out(dynamic(contentAdressed.drv))` to its content-adressed path
+    `ca(contentAdressed.drv)` which derives from
+    `sha256(out(dynamic(contentAdressed.drv)))`
+- We build `dependent.drv`
+  - We first compute `dynamic(dependent.drv)` to replace its
+    inputs by their real output path.
+    In that case, we replace `contentAdressed.drv!out` by
+    `ca(contentAdressed.drv)`
+  - We realise `dynamic(dependent.drv)`. This gives us an output path
+    `out(dynamic(dependent.drv))`
+- We build `transitivelyDependent.drv`
+  - We first compute `dynamic(transitivelyDependent.drv)` to replace its
+    inputs by their real output path.
+    In that case, that means replacing `dependent.drv!out` by
+    `out(dynamic(dependent.drv))`
+  - We realise `dynamic(transitivelyDependent.drv)`. This gives us an output path
+    `out(dynamic(transitivelyDependent.drv))`
+
+Now suppose that we slightly change the definition of `contentAdressed` in such
+a way that `contentAdressed.drv` will be modified, but its output will be the
+same. We try to rebuild the new `transitivelyDependent`. What happens is the
+following:
+
+- We instantiate the nix code, this gives us three new drv files:
+  `contentAdressed.drv`, `dependent.drv` and `transitivelyDependent.drv`
+- We build `contentAdressed.drv`.
+  - We first compute `dynamic(contentAdressed.drv)` to replace its
+    inputs by their real output path. Since there is none, we
+    have here `dynamic(contentAdressed.drv) == contentAdressed.drv`
+  - We realise `dynamic(contentAdressed.drv)`. This gives us an output path
+    `out(dynamic(contentAdressed.drv))`
+  - We compute `ca(contentAdressed.drv)` and notice that the
+    path already exists (since it's the same as the one we built previously),
+    so we discard the result.
+- We build `dependent.drv`
+  - We first compute `dynamic(dependent.drv)` to replace its
+    inputs by their real output path.
+    In that case, we replace `contentAdressed.drv!out` by
+    `ca(contentAdressed.drv)`
+  - We notice that `dynamic(dependent.drv)` is the same as before (since
+    `ca(contentAdressed.drv)` is the same as before), so we
+    just return the already existing path
+- We build `transitivelyDependent.drv`
+  - We first compute `dynamic(transitivelyDependent.drv)` to replace its
+    inputs by their real output path.
+    In that case, that means replacing `dependent.drv!out` by
+    `out(dynamic(dependent.drv))`
+  - Here again, we notice that `dynamic(transitivelyDependent.drv)` is the same as before,
+    so we don't build anything
+
+## nix-build process
```

Comment:
Some of this stuff sounds normative? If so, it should be moved outside of "exmples" back into "detailed design".